### PR TITLE
Jkmarx/upgrade node js

### LINKED ISSUE #1767: Upgrade Node.js to v8.x
[Node v8.0.0 was released on 2017-05-30](https://nodejs.org/en/blog/release/v8.0.0/). It includes npm version 5.0.0 which might help with the [intermittent build failures](https://github.com/npm/registry/issues/10) on TravisCI.

- [x] Upgrade puppet-nodejs to v4.0.0 

- [x] Update to nodejs v8 (npm v5 allows pinning)

puppet-nodejs Requires:
 Version 4.0.1 is compatible with:
Puppet Enterprise 2018.1.x, 2017.3.x, 2017.2.x, 2017.1.x, 2016.5.x, 2016.4.x
Puppet >=4.7.1 < 6.0.0

Ref #3096

### LINKED ISSUE #1945: Codecov for .js files fails in Travis CI?
### Steps to reproduce
- Look at  _any_ of our recent successful [Travis builds](https://travis-ci.org/refinery-platform/refinery-platform/builds)

### Observed behavior

**From the very end of Travis log:**
```
npm run codecov
npm ERR! Linux 4.4.0-83-generic
npm ERR! argv "/home/travis/.nvm/versions/node/v6.11.0/bin/node" "/home/travis/.nvm/versions/node/v6.11.0/bin/npm" "run" "codecov"
npm ERR! node v6.11.0
npm ERR! npm  v3.10.10
npm ERR! path /home/travis/build/refinery-platform/refinery-platform/refinery/package.json
npm ERR! code ENOENT
npm ERR! errno -2
npm ERR! syscall open
npm ERR! enoent ENOENT: no such file or directory, open '/home/travis/build/refinery-platform/refinery-platform/refinery/package.json'
npm ERR! enoent ENOENT: no such file or directory, open '/home/travis/build/refinery-platform/refinery-platform/refinery/package.json'
npm ERR! enoent This is most likely not a problem with npm itself
npm ERR! enoent and is related to npm not being able to find a file.
npm ERR! enoent 
npm ERR! Please include the following file with any support request:
npm ERR!     /home/travis/build/refinery-platform/refinery-platform/refinery/npm-debug.log
```

### Expected behavior
* For the `npm run codecov` to not yield an error

### PR DESCRIPTION
**Pull request by jkmarx**
- Resolves #1767
- Resolves #1945 
- Add feature to only record Cypress videos on fail.

~Overlaps with pull request #3410~

https://nodejs.org/en/about/releases/
https://nodejs.org/en/download/releases/
https://caniuse.com/#feat=es6

http://www.ecma-international.org/ecma-262/6.0/#sec-additions-and-changes-that-introduce-incompatibilities-with-prior-editions

____
https://github.com/nodejs/node/blob/master/doc/changelogs/CHANGELOG_V7.md
https://github.com/nodejs/node/blob/master/doc/changelogs/CHANGELOG_V8.md
https://github.com/nodejs/node/blob/master/doc/changelogs/CHANGELOG_V9.md
https://github.com/nodejs/node/blob/master/doc/changelogs/CHANGELOG_V10.md

### CODE REVIEW

File: deployment/puppet/Puppetfile.lock [hunk 1]
```diff
@@ -43,7 +42,7 @@ FORGE
       puppetlabs-stdlib (< 6.0.0, >= 4.13.1)
     puppetlabs-powershell (2.3.0)
     puppetlabs-reboot (2.1.2)
-    puppetlabs-stdlib (4.25.1)
+    puppetlabs-stdlib (5.2.0)
```
Comment (hackdna) [inline deployment/puppet/Puppetfile.lock]: Can we push it to 6.0.0 (the latest version) by any chance?
https://forge.puppet.com/puppetlabs/stdlib

File: refinery/ui/package.json [hunk 2]
```diff
@@ -18,9 +17,7 @@
     "url": "https://github.com/refinery-platform/refinery-platform/issues"
   },
   "devDependencies": {
-    "autoprefixer": "6.6.1",
-    "codecov.io": "0.1.6",
-    "cypress": "^3.1.5",
+    "cypress": "^3.3.1",
```
Comment (hackdna) [inline refinery/ui/package.json]: Can we push it to 3.4.0 (the latest version) by any chance?
https://www.npmjs.com/package/cypress

Comment (jkmarx) [reply refinery/ui/package.json]: Happens in pull request https://github.com/refinery-platform/refinery-platform/pull/3426

File: deployment/puppet/Puppetfile.lock (hunk 1 above)
Comment (jkmarx) [reply deployment/puppet/Puppetfile.lock]: Since it's a dependency, isn't it automatically resolved?

**Pull request merged**